ruler panic: runtime error: invalid memory address or nil pointer dereference

Noticed they were restarting more than expected:

```
$ kubectl get pod --namespace=cortex 
NAME                            READY     STATUS    RESTARTS   AGE
...
ruler-3796937216-etvy1          1/1       Running   47         4d
...
```

```
$ kubectl logs --namespace=cortex ruler-3796937216-etvy1 -p
time="2017-01-31T11:00:54Z" level=info msg="Ruler up and running" source="ruler.go:141" 
time="2017-01-31T11:01:55Z" level=warning msg="Error fetching from cache: read tcp 10.244.203.20:42064->10.244.204.99:11211: i/o timeout" source="chunk_store.go:455" 
time="2017-01-31T11:04:40Z" level=warning msg="Error fetching from cache: read tcp 10.244.203.20:33288->10.244.254.87:11211: i/o timeout" source="chunk_store.go:455" 
time="2017-01-31T11:05:11Z" level=warning msg="Error fetching from cache: read tcp 10.244.203.20:37170->10.244.253.92:11211: i/o timeout" source="chunk_store.go:455" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89870d]

goroutine 36436 [running]:
panic(0xab2fe0, 0xc420016050)
    /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).Send(0x0, 0xc422332d50, 0x1, 0x1)
    /go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:211 +0x4d
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).sendAlerts(0xc429cfe550, 0xc420432af0, 0x159f4340f7d, 0x159f4340f7d, 0xc42012f380)
    /go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:330 +0x5ae
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).Eval.func1(0xc4384e8740, 0xb79a11, 0x8, 0xc429cfe550, 0x159f4340f7d, 0x10f4e80, 0xc420432af0)
    /go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:275 +0xb52
created by github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).Eval
    /go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:301 +0x147
```

Also crashlooping:

```
time="2017-01-31T14:34:50Z" level=info msg="Removing stale ingester client for 10.244.229.104:9095" source="distributor.go:181"
2017/01/31 14:34:50 Failed to dial 10.244.229.104:9095: context canceled; please retry.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x89eacd]

goroutine 35945 [running]:
panic(0xab9140, 0xc420016050)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).Send(0x0, 0xc42149ef50, 0x1, 0x1)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:211 +0x4d
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).sendAlerts(0xc4224d2500, 0xc420010af0, 0x159f4f31c26, 0x159f4f31c26, 0xc4202e46c0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:330 +0x5ae
github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).Eval.func1(0xc42559c230, 0xb80777, 0x8, 0xc4224d2500, 0x159f4f31c26, 0x10ffa40, 0xc420010af0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:275 +0xb52
created by github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules.(*Group).Eval
	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/rules/manager.go:301 +0x147
```

--- a/ruler/ruler.go
+++ b/ruler/ruler.go
@@ -8,6 +8,7 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/log"
+	"github.com/prometheus/prometheus/notifier"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/rules"
 	"golang.org/x/net/context"
@@ -90,6 +91,7 @@ func (r *Ruler) newGroup(ctx context.Context, rs []rules.Rule) *rules.Group {
 		QueryEngine:    r.engine,
 		Context:        ctx,
 		ExternalURL:    r.alertURL,
+		Notifier:       notifier.New(&notifier.Options{}),
 	}
 	delay := 0 * time.Second // Unused, so 0 value is fine.
 	return rules.NewGroup("default", delay, rs, opts)

